source/Cosmos.Core/AdvancedObject.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Cosmos.Core;
+using IL2CPU.API;
+
+namespace Cosmos.Core
+{
+    /// <summary>
+    /// This Class is created for Add more methods to objects by add this to base class
+    /// </summary>
+    public class AdvancedObject
+    {
+        public unsafe uint GetTypeId()
+        {
+            return GCImplementation.GetType(this);
+        }
+        /// <summary>
+        /// If you stuck on a problem run this function
+        /// <c>Waring: Runnig this function may get to crash</c>
+        /// </summary>
+        public void FixStackPoint()
+        {
+            if (NeedToFixStackPoint())
+            {
+                Global.mDebugger.Send("Stack is need to be fixed");
+                FixStackPointTo(GetTypeId());
+            }
+        }
+
+        /// <summary>
+        /// This function is set the stack point to the target type
+        /// <c>Waring: Runnig this function may get to crash</c>
+        /// </summary>
+        /// <param name="stack">The target type id</param>
+        public void FixStackPointTo(uint stack)
+        {
+            CPU.DisableInterrupts();
+            CPU.SetESPValue(stack);
+            CPU.EnableInterrupts();
+        }
+
+        /// <summary>
+        /// Check if the stack type is match to the object type
+        /// </summary>
```

Comment:
In what situation should the stack position be equal to the object type?

Reply:
Sometimes that i run an OS with the framework I Got a proplem with VTableImpl "Method Not Found", when i got the problem it was ESP register with worng type id value

Reply:
Could you open an issue with the an example where this is happening?

Reply:
I opened an isssue